### PR TITLE
2.5 Host Registration changes

### PR DESCRIPTION
**Pull request by stejskalleos**
https://github.com/theforeman/foreman-documentation/issues/513

### CODE REVIEW

**Comment by stejskalleos** [comment]
@melcorr

**Comment by melcorr** [comment]
Hi @suyogsainkar - can you give this a sanity check for grammar/style and make any suggestions needed? Thanks

File: guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc [hunk 1]
```diff
@@ -4,11 +4,13 @@
...
 With this registration method you can register hosts to {Project} by generating a `curl` command on {Project} and running this command on hosts.
-This method uses the global registration template that gives you complete control over the process of the host registration.
+This method uses two templates: `global registration` template and `host initial configuration` template. That gives you complete control over the process of the host registration.
+Default templates can be set in the Settings > provisioning.
+
 
 ifndef::satellite[]
 Note that you can extend the parameters by plug-ins.
-For more information, see https://github.com/theforeman/foreman/blob/develop/developer_docs/how_to_create_a_plugin.asciidoc[How to Create a Plugin].
+For more information, see https://github.com/theforeman/foreman/blob/develop/developer_docs/how_to_create_a_plugin.asciidoc[How to Create a Plugin] and http://theforeman.github.io/foreman/?path=/docs/introduction-slot-and-fill--page.asciidoc[Slot and Fill]
```
Comment (lzap) [inline guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc:13]: Nitpick: the URL does not open the correct chapter, but it opens the page.

Comment (stejskalleos) [reply guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc:13]: Really? For me it opens _Slot and Fill_ page correctly: http://theforeman.github.io/foreman/?path=/docs/introduction-slot-and-fill--page

**Comment by stejskalleos** [comment]
@melcorr @suyogsainkar 

Since Satellite 6.9 Global registration is default registration method, so I added deprecation info at page with bootstrap.py registration. Not sure if it is enough, maybe some page relocation would be better idea ...

File: guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc [hunk 2]
```diff
@@ -4,11 +4,13 @@
 Use the following procedure to register a host to {ProjectNameX}.
 
 With this registration method you can register hosts to {Project} by generating a `curl` command on {Project} and running this command on hosts.
-This method uses the global registration template that gives you complete control over the process of the host registration.
+This method uses two templates: `global registration` template and `host initial configuration` template. That gives you complete control over the process of the host registration.
```
Comment (suyogsainkar) [inline guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc:7]: over the host registration process.

File: guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc [hunk 3]
```diff
@@ -52,32 +34,54 @@ Optional: If the *Registration* feature is not enabled on your {SmartProxy}, ent
...
+Fields that inherits value from *Host group*: Operating System, Activation Key(s) and Lifecycle environment.
 
 . From the *Operating System* list, select the operating system of hosts that you want to register.
 
-. From the *{SmartProxy}* list, select the {SmartProxy} to register hosts through. You must select the internal {SmartProxy} if you do not want to use an external {SmartProxy}.
+. Optional: From the *{SmartProxy}* list, select the {SmartProxy} to register hosts through. You must select the internal {SmartProxy} if you do not want to use an external {SmartProxy}.
+
+. Optional: *Insecure* - This makes the first call insecure, however, during this first call, hosts download the CA file from {Project}. Hosts will use this CA file to connect to {Project} with all future calls making them secure.
++
+If an attacker, located in the network between {Project} and a host, fetches the CA file from the first insecure call, the attacker will be able to access the content of the API calls to and from the registered host and the JWT.
```
Comment (suyogsainkar) [inline guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc:49]: JSON Web Tokens (JWT)

**Review by suyogsainkar — CHANGES_REQUESTED** [review]
@stejskalleos thanks for the updates. I've added a few comments.

File: guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc [hunk 4]
```diff
@@ -4,11 +4,13 @@
 Use the following procedure to register a host to {ProjectNameX}.
 
 With this registration method you can register hosts to {Project} by generating a `curl` command on {Project} and running this command on hosts.
-This method uses the global registration template that gives you complete control over the process of the host registration.
+This method uses two templates: `global registration` template and `host initial configuration` template. That gives you complete control over the process of the host registration.
+Default templates can be set in the Settings > provisioning.
```
[the remark this answers is not in the document]
Comment (stejskalleos) [reply guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc:8]: > You need to clone the default templates and edit the clones.

Yeah but that is not necessary for the registration process. That's only for the users who wants to edit the templates

**Comment by stejskalleos** [comment]
@suyogsainkar Docs update

I made some more changes (forget to update some doc files), if you have any questions feel free to ping me on IRC.

**Comment by melcorr** [comment]
@suyogsainkar any updates that you can provide to unblock this?

**Comment by melcorr** [comment]
@suyogsainkar do you plan to return to this? It's been a long time. I will just merge as is if I don't hear any feedback in the next two days.

**Comment by melcorr** [comment]
I am merging this. 
Any additional corrections can be done in a new PR. 
Thank you very much @stejskalleos for your comprehensive docs.

File: guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc [hunk 5]
```diff
@@ -1,14 +1,14 @@
-[id="registering-a-host-to-project-using-the-global-registration-template_{context}"]
-= Registering a Host to {ProjectName} Using the Global Registration Template
+[id="registering-a-host_{context}"]
+= Registering a Host to {ProjectName}
```
Comment (lzap) [inline guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc:2]: For the record, this PR changed the ID however the code still contained references:

```
$ ag "registering-a-host-to-project-using-the-global-registration-template"
guides/common/assembly_configuring-and-setting-up-remote-jobs.adoc
32:For more information, see {ManagingHostsDocURL}registering-a-host-to-project-using-the-global-registration-template_managing-hosts[Registering a Host to {ProjectName} Using the Global Registration Template].

guides/doc-Content_Management_Guide/topics/QuickStart.adoc
154:Use your activation key to register a host to the Content View. For more information on registering a host, see {ManagingHostsDocUR}registering-a-host-to-project-using-the-global-registration-template_managing-hosts

guides/doc-Managing_Hosts/topics/Registering_Hosts.adoc
25:* xref:registering-a-host-to-project-using-the-global-registration-template_managing-hosts[]

guides/doc-Managing_Hosts/topics/using_insights_with_satellite_hosts.adoc
12:For more information, see {ManagingHostsDocURL}registering-a-host-to-project-using-the-global-registration-template_managing-hosts[Registering a Host to {ProjectName} Using the Global Registration Template] in the _Managing Hosts_ guide.
```

I wonder how was possible that our HTML validation passed. I mean, those that has links to the site surely passed, this is something we need to improve, but I would expect `xref:registering-a-host-to-project-using-the-global-registration-template_managing-hosts[]` to fail.

**Comment by lzap** [comment]
Also I do not understand why the ID was changed to "registering-a-host_CONTEXT". Now we have two such IDs, one with and one without CONTEXT. Was this intentional?

```
guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-satellite-manually.adoc
1:[id="registering-a-host"]
2:[id="registering-a-host-to-satellite"]

guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc
1:[id="registering-a-host_{context}"]
```

**Pull request merged**